File: Python/Insert Interval.py
# Definition for an interval.
# class Interval:
#     def __init__(self, s=0, e=0):
#         self.start = s
#         self.end = e

class Solution:
    # @param intervals, a list of Intervals
    # @param newInterval, a Interval
    # @return a list of Interval
    def insert(self, intervals, newInterval):
        intervals.append(newInterval)
        intervals.sort(key=lambda x: x.start)
        res = []
        for i in intervals:
            if res and i.start <= res[-1].end:
                res[-1].end = max(i.end, res[-1].end)
            else:
                res.append(i)
        return res       
        

File: Java/src/Insert Interval.java
import java.util.ArrayList;

/**
 * Definition for an interval.
 * public class Interval {
 * int start;
 * int end;
 * Interval() { start = 0; end = 0; }
 * Interval(int s, int e) { start = s; end = e; }
 * }
 */
public class Solution
{
  public List<Interval> insert(List<Interval> intervals, Interval newInterval)
  {
    List<Interval> result = new ArrayList<>();
    int i = 0;
    while (i < intervals.size())
    {
      if (newInterval.start <= intervals.get(i).end)
      {
        if (newInterval.end < intervals.get(i).start)
        {
          break;
        }
        newInterval.start = Math.min(newInterval.start, intervals.get(i).start);
        newInterval.end = Math.max(newInterval.end, intervals.get(i).end);
      }
      else
      {
        result.add(intervals.get(i));
      }
      i++;
    }
    result.add(newInterval);
    while (i < intervals.size())
    {
      result.add(intervals.get(i));
      i++;
    }

    return result;
  }
}
